pkg/reconciler/kubernetes/tektonhub/tektonhub.go

```diff
@@ -0,0 +1,595 @@
+		return nil, errKeyMissing
+	}
+
+	return secret, nil
+}
+
+func (r *Reconciler) getConfigMapForHub(ctx context.Context, name, targetNs string, keys []string) (*corev1.ConfigMap, error) {
+	configMap, err := r.kubeClientSet.CoreV1().ConfigMaps(targetNs).Get(ctx, name, metav1.GetOptions{})
+	if err != nil {
+		return nil, err
+	}
+
+	allKeys := true
+	for _, key := range keys {
+		if _, ok := configMap.Data[key]; !ok {
+			allKeys = false
+			break
+		}
+	}
+
+	if !allKeys {
+		return nil, errKeyMissing
+	}
+
+	return configMap, nil
+}
+
+func createApiConfigMap(name, namespace string, th *v1alpha1.TektonHub) *corev1.ConfigMap {
+	return &corev1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: namespace,
+			Labels: map[string]string{
+				"app": "api",
+			},
+		},
+		Data: map[string]string{
+			"CONFIG_FILE_URL": th.Spec.Api.HubConfigUrl,
+		},
+	}
+}
+
+func createDbSecret(name, namespace string, existingSecret *corev1.Secret) *corev1.Secret {
+	s := &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: namespace,
+			Labels: map[string]string{
+				"app": "db",
+			},
+		},
+		Type: corev1.SecretTypeOpaque,
+	}
+
+	if existingSecret != nil && existingSecret.Data != nil {
+		s.Data = existingSecret.Data
+	}
+
+	s.StringData = make(map[string]string)
+
+	if s.Data["POSTGRES_DB"] == nil || len(s.Data["POSTGRES_DB"]) == 0 {
+		s.StringData["POSTGRES_DB"] = "hub"
+	}
+
+	if s.Data["POSTGRES_USER"] == nil || len(s.Data["POSTGRES_USER"]) == 0 {
+		s.StringData["POSTGRES_USER"] = "hub"
+	}
+
+	if s.Data["POSTGRES_PASSWORD"] == nil || len(s.Data["POSTGRES_PASSWORD"]) == 0 {
+		s.StringData["POSTGRES_PASSWORD"] = "hub"
+	}
+
+	if s.Data["POSTGRES_PORT"] == nil || len(s.Data["POSTGRES_PORT"]) == 0 {
+		s.StringData["POSTGRES_PORT"] = "5432"
+	}
+
+	return s
+}
+
+func (r *Reconciler) applyManifest(ctx context.Context, manifestLocation string, th *v1alpha1.TektonHub, installerSetName, version, prefixName string) error {
+	manifest := r.manifest.Append()
+	logger := logging.FromContext(ctx)
+
+	if err := common.AppendManifest(&manifest, manifestLocation); err != nil {
+		return err
+	}
+
+	manifest = manifest.Filter(mf.Not(mf.Any(mf.ByKind("Secret"), mf.ByKind("Namespace"), mf.ByKind("ConfigMap"))))
+
+	manifest, err := manifest.Transform(
+		mf.InjectOwner(th),
+		mf.InjectNamespace(namespace),
+	)
+	if err != nil {
+		logger.Error("failed to transform manifest")
+		return err
+	}
+
+	if err := createInstallerSet(ctx, r.operatorClientSet, th, manifest,
+		version, installerSetName, prefixName, namespace); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (r *Reconciler) installed(ctx context.Context, instance v1alpha1.TektonComponent) (*mf.Manifest, error) {
+	// Create new, empty manifest with valid client and logger
+	installed := r.manifest.Append()
+	stages := common.Stages{common.AppendInstalled}
+	err := stages.Execute(ctx, &installed, instance)
+	return &installed, err
+}
```

Comment:
imo, we do not need this. we should just delete the installerset and let installerSet reconciler worry about getting items and deleting them.

In other words, the logic in this function is clusterOperation related not component lifecycle related. So let us not handle it here.